scipy/stats/tests/test_qmc.py

```diff
@@ -0,0 +1,810 @@
+import copy
+from collections import Counter
+
+import pytest
+import numpy as np
+from numpy.testing import (assert_allclose, assert_almost_equal, assert_,
+                           assert_equal, assert_array_almost_equal,
+                           assert_array_equal)
+from scipy.stats import shapiro
+
+from scipy.stats._sobol import _test_find_index
+from scipy.stats import qmc
+from scipy.stats._qmc import (van_der_corput, n_primes, primes_from_2_to,
+                              _perturb_discrepancy, _update_discrepancy,
+                              QMCEngine)
+
+
+class TestUtils:
+    def test_scale(self):
+        # 1d scalar
+        space = [[0], [1], [0.5]]
+        out = [[-2], [6], [2]]
+        scaled_space = qmc.scale(space, l_bounds=-2, u_bounds=6)
+
+        assert_allclose(scaled_space, out)
+
+        # 2d space
+        space = [[0, 0], [1, 1], [0.5, 0.5]]
+        bounds = np.array([[-2, 0], [6, 5]])
+        out = [[-2, 0], [6, 5], [2, 2.5]]
+
+        scaled_space = qmc.scale(space, l_bounds=bounds[0], u_bounds=bounds[1])
+
+        assert_allclose(scaled_space, out)
+
+        scaled_back_space = qmc.scale(scaled_space, l_bounds=bounds[0],
+                                      u_bounds=bounds[1], reverse=True)
+        assert_allclose(scaled_back_space, space)
```

Comment:
This is probably fine, as the function is so simple, but if it were more complicated I would prefer that a round trip test be with randomly generated data so that it's easier to tell that the inputs are not just lucky or specifically chosen to pass tests. Consider replacing with something like:
```python3
np.random.seed(0)
sample = np.random.rand(30, 10)
a = -np.random.rand(10) * 10
b = np.random.rand(10) * 10
scaled = qmc.scale(sample, a, b, reverse=False)
unscaled = qmc.scale(scaled, a, b, reverse=True)
assert_allclose(unscaled, sample)
```

Reply:
I added this to the test suite.